### PR TITLE
Automating package releases

### LINKED ISSUE #46: setup automatic releasing of the packages
Similar to https://github.com/std-uritemplate/std-uritemplate/blob/f8454a675e2ae705908f94bb49df8f63673727cf/.github/workflows/release.yml#L289

Key points:
- uses OIDC
- triggers on tag/release

For the initial publish we'll need to do that locally, transfer the packages to the [verified publisher](https://pub.dev/publishers/graph.microsoft.com/packages), and ultimately add permissions for the repo to publish the packages

### PR DESCRIPTION
**Pull request by ricardoboss**
Fixes #46

Based on #60 (that's why this PR targets that branch)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+jobs:
+  publish-gh-releases:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Release
+        uses: softprops/action-gh-release@v2
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+      # We faced problem like: https://github.com/cli/cli/issues/6599
```
Comment (baywet) [inline .github/workflows/release.yml:20]: the issue is marked as fixed, maybe this is not necessary anymore?

Comment (ricardoboss) [reply .github/workflows/release.yml]: I removed that part. We'll see if it works when this first runs.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+
+      # Changelog is generated from GH releases
+      - name: Generate Changelog
+        uses: rhysd/changelog-from-release/action@v3
+        with:
+          file: CHANGELOG.md
+          commit: false
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (baywet) [inline .github/workflows/release.yml:28]: why is this step using a different secret than the one at line 20?

**Review by baywet — APPROVED** [review]
Thanks for the contribution!

One question left though

File: .github/workflows/release.yml (hunk 2 above)
Comment (ricardoboss) [reply .github/workflows/release.yml:28]: I noticed that too, but thought it might just be an alias. We should probably be consistent here

**Review by baywet — APPROVED** [review]
Thank you for making the changes!

**Pull request merged**